c/src/ml-api-service.c

```diff
@@ -344,14 +474,7 @@ ml_service_new (const char *config, ml_service_h * handle)
 
   object = json_node_get_object (root);
 
-  if (json_object_has_member (object, "single") ||
-      json_object_has_member (object, "pipeline")) {
-    service_type = ML_SERVICE_TYPE_EXTENSION;
-  } else {
-    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-        "Failed to parse configuration file, cannot get the valid type from configuration.");
-  }
-
+  service_type = _ml_service_get_type (object);
```

Comment:
need handle error case if service type is ML_SERVICE_TYPE_UNKNOWN

Reply:
Thanks, updated!